ctyparser/bigcty.py

```diff
@@ -92,32 +114,56 @@ def import_dat(self, dat_file: Union[str, os.PathLike]) -> None:
             last = ''
+            
             while True:
+                # read a line
                 line = file.readline().rstrip('\x0D').strip(':')
                 if not line:
                     break
+                # check if the line introduces new DXCC (line is starting with character)
+                # samples:
+                # Switzerland:              14:  28:  EU:   46.87:    -8.12:    -1.0:  HB:
+                # Sicily:                   15:  28:  EU:   37.50:   -14.00:    -1.0:  *IT9:
                 if line != '' and line[0].isalpha():
+                    # split line by ':' and remove spaces
                     segments = [x.strip() for x in line.split(':')]
+                    # check if last segment starts with '*'
+                    # A “*” preceding this prefix indicates that the country is on the DARC WAEDC 
+                    # list, and counts in CQ-sponsored contests, but not ARRL-sponsored contests
                     if segments[7][0] == '*':
+                        # remove '*' and add a marker to country name
                         segments[7] = segments[7][1:]
                         segments[0] += ' (not DXCC)'
+                    # store data in dict
                     cty_dict[segments[7]] = {'entity': segments[0], 'cq': int(segments[1]),
                                              'itu': int(segments[2]), 'continent': segments[3],
                                              'lat': float(segments[4]), 'long': float(segments[5]),
                                              'tz': -1*float(segments[6]), 'len': len(segments[7]),
-                                             'primary_pfx': segments[7]}
+                                             'primary_pfx': segments[7], 'exact_match': False}
+                    # store country name, which is key of cty_dict for use with continued data
                     last = segments[7]
 
+                # check if the line continues DXCC (line is starting with space)
+                # samples:
+                # IB9,ID9,IE9,IF9,II9,IJ9,IO9,IQ9,IR9,IT9,IU9,IW9,IY9,=II0GDF/9,=IQ1QQ/9,=IQ6KX/9,=IT9ACJ/I/BO,
+                # =IT9YBL/SG,=IT9ZSB/LH,=IW0HBY/9;
                 elif line != '' and line[0].isspace():
+                    # remove spaces, trailing separators and split by ','
                     overrides = line.strip().rstrip(';').rstrip(',').split(',')
+                    
                     for item in overrides:
+                        # check if prefix/call is not already in dict
                         if item not in cty_dict.keys():
-                            data = cty_dict[last]
+                            # get the already stored data from primary prefix
+                            data = dict(cty_dict[last])
+                            # apply regex to extract the prefix and overrides
                             match = re.search(self.regex_dat, item)
```

Comment:
Same comment about commenting.

Casting `cty_dict[last]` to `dict` is totally unnecessary because `cty_dict` is already of type `Dict[str, Dict]`.

```suggestion
                            data = cty_dict[last]
                            match = re.search(self.regex_dat, item)
```

Reply:
okay, here I should have commented the 'why' ;-)
dict is necessary to create a new dict instance, not to cast. Otherwise you will override the data of the primary prefix if e.g. cq zone is changed by the sub prefixes.
Please try this testdata with and without the dict:
```
China:                    24:  44:  AS:   36.00:  -102.00:    -8.0:  BY:
    3H0(23)[42];
```
with dict:
```
{'BY': {'entity': 'China', 'cq': 24, 'itu': 44, 'continent': 'AS', 'lat': 36.0, 'long': -102.0, 'tz': 8.0, 'len': 2, 'primary_pfx': 'BY', 'exact_match': False}, 
'3H0': {'entity': 'China', 'cq': 23, 'itu': 42, 'continent': 'AS', 'lat': 36.0, 'long': -102.0, 'tz': 8.0, 'len': 2, 'primary_pfx': 'BY', 'exact_match': False}}
```
Please look at itu and cq, and now without dict:
```
{'BY': {'entity': 'China', 'cq': 23, 'itu': 42, 'continent': 'AS', 'lat': 36.0, 'long': -102.0, 'tz': 8.0, 'len': 2, 'primary_pfx': 'BY', 'exact_match': False}, 
'3H0': {'entity': 'China', 'cq': 23, 'itu': 42, 'continent': 'AS', 'lat': 36.0, 'long': -102.0, 'tz': 8.0, 'len': 2, 'primary_pfx': 'BY', 'exact_match': False}}

```
The cq and itu was overwritten

Reply:
Ah alright, then what we want to do is `copy.deepcopy(cty_dict[last])`, which will ensure all data is passed by reference.

https://docs.python.org/3/library/copy.html

Reply:
Okay, I changed it accordingly